src/crimson/osd/ops_executer.h

```diff
@@ -166,16 +166,22 @@ class OpsExecuter : public seastar::enable_lw_shared_from_this<OpsExecuter> {
 
   Ref<PG> pg; // for the sake of object class
   ObjectContextRef obc;
+  ObjectContextRef clone_obc; // if we create a clone
   const OpInfo& op_info;
   ceph::static_ptr<ExecutableMessage,
                    sizeof(ExecutableMessagePimpl<void>)> msg;
   std::optional<osd_op_params_t> osd_op_params;
   bool user_modify = false;
+  bool head_existed = false;
   ceph::os::Transaction txn;
 
   size_t num_read = 0;    ///< count read ops
   size_t num_write = 0;   ///< count update ops
 
+  SnapContext snapc; // writer snap context
+  // obc will hold both (old/resulting)obs
```

Comment:
I'm not sure this context is useful.  obc unconditionally references the snapcontext even for a read, right?  This also doesn't seem to be a good place to explain that the obc contains the obs.